nrfx_config_nrf9160.h

```diff
@@ -36,243 +36,124 @@
-#define NRF_EGU0 \
-        ((NRF_EGU_Type *)DT_NORDIC_NRF_EGU_EGU_0_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_EGU_EGU_1_BASE_ADDRESS
-#define NRF_EGU1 \
-        ((NRF_EGU_Type *)DT_NORDIC_NRF_EGU_EGU_1_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_EGU_EGU_2_BASE_ADDRESS
-#define NRF_EGU2 \
-        ((NRF_EGU_Type *)DT_NORDIC_NRF_EGU_EGU_2_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_EGU_EGU_3_BASE_ADDRESS
-#define NRF_EGU3 \
-        ((NRF_EGU_Type *)DT_NORDIC_NRF_EGU_EGU_3_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_EGU_EGU_4_BASE_ADDRESS
-#define NRF_EGU4 \
-        ((NRF_EGU_Type *)DT_NORDIC_NRF_EGU_EGU_4_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_EGU_EGU_5_BASE_ADDRESS
-#define NRF_EGU5 \
-        ((NRF_EGU_Type *)DT_NORDIC_NRF_EGU_EGU_5_BASE_ADDRESS)
-#endif
-
-#ifdef DT_INST_0_NORDIC_NRF_FICR_BASE_ADDRESS
-#define NRF_FICR \
-        ((NRF_FICR_Type *)DT_INST_0_NORDIC_NRF_FICR_BASE_ADDRESS)
-#endif
-
-#ifdef DT_NORDIC_NRF_GPIO_GPIO_0_BASE_ADDRESS
-#define NRF_P0 \
-        ((NRF_GPIO_Type *)DT_NORDIC_NRF_GPIO_GPIO_0_BASE_ADDRESS)
-#endif
-
-#ifdef DT_NORDIC_NRF_GPIOTE_GPIOTE_0_BASE_ADDRESS
-#define NRF_GPIOTE \
-        ((NRF_GPIOTE_Type *)DT_NORDIC_NRF_GPIOTE_GPIOTE_0_BASE_ADDRESS)
-#endif
-
-#ifdef DT_NORDIC_NRF_I2S_I2S_0_BASE_ADDRESS
-#define NRF_I2S \
-        ((NRF_I2S_Type *)DT_NORDIC_NRF_I2S_I2S_0_BASE_ADDRESS)
-#endif
-
-#ifdef DT_INST_0_NORDIC_NRF_KMU_BASE_ADDRESS
-#define NRF_KMU \
-        ((NRF_KMU_Type *)DT_INST_0_NORDIC_NRF_KMU_BASE_ADDRESS)
-#endif
-
-#ifdef DT_INST_0_NORDIC_NRF91_FLASH_CONTROLLER_BASE_ADDRESS
-#define NRF_NVMC \
-        ((NRF_NVMC_Type *)DT_INST_0_NORDIC_NRF91_FLASH_CONTROLLER_BASE_ADDRESS)
-#endif
-
-#ifdef DT_NORDIC_NRF_PDM_PDM_0_BASE_ADDRESS
-#define NRF_PDM \
-        ((NRF_PDM_Type *)DT_NORDIC_NRF_PDM_PDM_0_BASE_ADDRESS)
-#endif
-
-#ifdef DT_INST_0_NORDIC_NRF_POWER_BASE_ADDRESS
-#define NRF_POWER \
-        ((NRF_POWER_Type *)DT_INST_0_NORDIC_NRF_POWER_BASE_ADDRESS)
-#endif
-
-#ifdef DT_NORDIC_NRF_PWM_PWM_0_BASE_ADDRESS
-#define NRF_PWM0 \
-        ((NRF_PWM_Type *)DT_NORDIC_NRF_PWM_PWM_0_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_PWM_PWM_1_BASE_ADDRESS
-#define NRF_PWM1 \
-        ((NRF_PWM_Type *)DT_NORDIC_NRF_PWM_PWM_1_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_PWM_PWM_2_BASE_ADDRESS
-#define NRF_PWM2 \
-        ((NRF_PWM_Type *)DT_NORDIC_NRF_PWM_PWM_2_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_PWM_PWM_3_BASE_ADDRESS
-#define NRF_PWM3 \
-        ((NRF_PWM_Type *)DT_NORDIC_NRF_PWM_PWM_3_BASE_ADDRESS)
-#endif
-
-#ifdef DT_INST_0_NORDIC_NRF_REGULATORS_BASE_ADDRESS
-#define NRF_REGULATORS \
-        ((NRF_REGULATORS_Type *)DT_INST_0_NORDIC_NRF_REGULATORS_BASE_ADDRESS)
-#endif
-
-#ifdef DT_NORDIC_NRF_RTC_RTC_0_BASE_ADDRESS
-#define NRF_RTC0 \
-        ((NRF_RTC_Type *)DT_NORDIC_NRF_RTC_RTC_0_BASE_ADDRESS)
-#endif
-#ifdef DT_NORDIC_NRF_RTC_RTC_1_BASE_ADDRESS
-#define NRF_RTC1 \
-        ((NRF_RTC_Type *)DT_NORDIC_NRF_RTC_RTC_1_BASE_ADDRESS)
-#endif
+#if defined(CONFIG_ARM_NONSECURE_FIRMWARE)
```

Comment:
You could've written a macro:

```
#define NRF_PERIPH(P) #if defined(CONFIG_ARM_NONSECURE_FIRMWARE) \
                                            #define P P##_NS \
                                      #else \
                                           #define P P##_S \
                                      #endif

NRF_PERIPH(NRF_CLOCK);
NRF_PERIPH(NRF_DPPIC);
...
```
That would avoid duplication of the lists.
What do you think?

Reply:
Yes, I think it could work.

We've got one exception: `NRF_GPIOTE` that needs to be mapped to `NRF_GPIOTE1_NS` or `NRF_GPIOTE0_S`, but it can be handled separately (it will be more visible then, what is good).

Another thing is that some peripherals are accessible only from secure code, so only `*_S` symbols are defined for them. But this also should not be a problem. The only impact I can see is that it would not be possible to use something like `#if defined(NRF_CRYPTOCELL)` to check if the CRYPTOCELL peripheral is available, because `NRF_CRYPTOCELL` will be defined also for non-secure firmware. But we don't do such checks.

I agree to apply such modification. Any objections from others?

Reply:
Yep, just use special casing for those that are not symmetric in the secure vs non-secure worlds.